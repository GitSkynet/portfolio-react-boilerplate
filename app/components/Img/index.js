/**
 * 1. Fluid images for responsive purposes.
 * 2. Offset `alt` text from surrounding copy.
 * 3. Setting `vertical-align` removes the whitespace that appears under `img`
 *    elements when they are dropped into a page as-is. Safer alternative to
 *    using `display: block;`.
 */

import styled from 'styled-components';

const Img = styled.img`
  width: 100%; /* [1] - Internet Explorer */
  max-width: 100%; /* [1] */
  font-style: italic; /* [2] */
  vertical-align: middle; /* [3] */
  
  &[height],
  &[width] {
    width: unset;
    max-width: none;
  }
`;
export default Img;
