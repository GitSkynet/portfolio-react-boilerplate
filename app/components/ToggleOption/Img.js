import styled from 'styled-components';

import NormalImg from '../../components/Img';

const Img = styled(NormalImg)`
  filter: saturate(100%);
  transition: all 0.2s;
`;

export default Img;
