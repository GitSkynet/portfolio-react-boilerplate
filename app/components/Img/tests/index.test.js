import React from 'react';
import { shallow } from 'enzyme';

import Img from '../index';

describe('<Img />', () => {
  it('should render and match the snapshot', () => {
    const renderedComponent = shallow(<Img />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
