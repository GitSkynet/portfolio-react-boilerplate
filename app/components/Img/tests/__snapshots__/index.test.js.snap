// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Img /> should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Img__Img />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "Img__Img-eVCnLo vQDkP",
    },
    "ref": null,
    "rendered": null,
    "type": "img",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "Img__Img-eVCnLo vQDkP",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
